### ISSUE TITLE
PYI049: false positive for private inner class

### ISSUE THREAD

**Issue by arvidfm** [astral-sh/ruff#9948]
The following code results in a PYI049 error even though `_B` is very much used:

```py
from typing import TypedDict


class A:
    class _B(TypedDict):
        pass

    def f(self) -> None:
        print(A._B())


A().f()
```

```console
$ ruff check --select=PYI test.py
test.py:5:11: PYI049 Private TypedDict `_B` is never used
Found 1 error.
$ ruff --version
ruff 0.2.1
```

**Comment by charliermarsh** [comment]
We should avoid flagging these when they're defined in a class scope. @AlexWaygood just because I see you on this issue, do you want to fix? Otherwise I can.

**Comment by AlexWaygood** [comment]
> @AlexWaygood just because I see you on this issue, do you want to fix? Otherwise I can.

I'm super busy this week at work -- I have a lot of handover stuff to be getting on with :( If you could take a look, that would be great!

**Comment by charliermarsh** [comment]
No prob, on it.
